### DISCUSSION TITLE
Applying effects synchronously instead of consecutively

### DISCUSSION THREAD

**Discussion by psuskin** [DBraun/DawDreamer#159] [Q&A]
I've noticed that the runtime of my script increases more or less proportionally to the number of plugins in a given graph. However, DAWs are generally capable of applying the effects of multiple plugins synchronously, so the runtime of the render process shouldn't need to go up much, if at all. Am I setting up my graph incorrectly? I've tried two variants of graphs, both of which take equally long:
`[(playback/synth, []), (plugin1, [playback/synth]), (plugin2, [plugin2]), ...]`
`[(playback/synth, []), (plugin1, [playback/synth]), (plugin2, [playback/synth]), ...]`
Or does the `render` function intend to apply effects consecutively? Because in this case, I'd like to contribute by finding a way to apply effects synchronously.

**Comment by DBraun** [comment]
Sorry I missed this question earlier. We’re at the mercy of the JUCE AudioProcessorGraph class, so I wouldn’t expect anything to be multithreaded within a single RenderEngine. In other words only one processor will be evaluated at a time.
